pygbm/gradient_boosting.py

```diff
@@ -244,6 +242,10 @@ def fit(self, X, y):
                 _update_raw_predictions(leaves_data, raw_predictions[:, k])
                 toc_pred = time()
                 acc_prediction_time += toc_pred - tic_pred
+                # make sure we collect memory
+                del grower
```

Comment:
Maybe we should only collect the last grower, after the main loop?

Reply:
Unfortunately, that would not solve it fully, the idea is:
 * I have 20GB of RAM used, the grower takes 8GB, 28GB used
 * next loop, first the new grower allocates 8GB, I have 36GB used
 * old grower goes out of scope, somewhere in the future it will be garbage collected

Even without the explicit gc.collect, the 36GB RAM situation can still occur, hope I've explained myself well.